File: Java/Android/OpenGLES/source/core/src/loon/action/sprite/Label.java
package loon.action.sprite;

import loon.core.LObject;
import loon.core.geom.RectBox;
import loon.core.graphics.LColor;
import loon.core.graphics.LFont;
import loon.core.graphics.opengl.GLEx;
import loon.core.graphics.opengl.LTexture;


/**
 * Copyright 2008 - 2009
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 * 
 * @project loon
 * @author cping
 * @email：javachenpeng@yahoo.com
 * @version 0.1
 */
public class Label extends LObject implements ISprite {

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	private LFont font;

	private boolean visible;

	private int width, height;

	private LColor color;

	private String label;

	public Label(String label, int x, int y) {
		this(LFont.getDefaultFont(), label, x, y);
	}

	public Label(String label, String font, int type, int size, int x, int y) {
		this(LFont.getFont(font, type, size), label, x, y);
	}

	public Label(LFont font, String label, int x, int y) {
		this.font = font;
		this.label = label;
		this.color = LColor.black;
		this.visible = true;
		this.setLocation(x, y);
	}

	public void setFont(String fontName, int type, int size) {
		setFont(LFont.getFont(fontName, type, size));
	}

	public void setFont(LFont font) {
		this.font = font;
	}

	@Override
	public void createUI(GLEx g) {
		if (visible) {
			LFont oldFont = g.getFont();
			int oldColor = g.getColorRGB();
			g.setFont(font);
			g.setColor(color);
			this.width = font.stringWidth(label);
			this.height = font.getSize();
			if (alpha > 0 && alpha < 1) {
				g.setAlpha(alpha);
				g.drawString(label, x(), y());
				g.setAlpha(1.0F);
			} else {
				g.drawString(label, x(), y());
			}
			g.setFont(oldFont);
			g.setColor(oldColor);
		}
	}


	@Override
	public int getWidth() {
		return width;
	}

	@Override
	public int getHeight() {
		return height;
	}

	@Override
	public void update(long timer) {

	}

	@Override
	public RectBox getCollisionBox() {
		return getRect(x(), y(), width, height);
	}

	@Override
	public boolean isVisible() {
		return visible;
	}

	@Override
	public void setVisible(boolean visible) {
		this.visible = visible;
	}

	public String getLabel() {
		return label;
	}

	public void setLabel(int label) {
		setLabel(String.valueOf(label));
	}

	public void setLabel(String label) {
		this.label = label;
	}

	public LColor getColor() {
		return color;
	}

	public void setColor(LColor color) {
		this.color = color;
	}

	@Override
	public LTexture getBitmap() {
		return null;
	}

	@Override
	public void dispose() {

	}

}

File: C#/WindowsPhone/LGame-XNA-lib/Loon.Action.Sprite/Label.cs
using Loon.Core;
using Loon.Core.Graphics;
using Loon.Core.Graphics.Opengl;
using Loon.Core.Geom;
namespace Loon.Action.Sprite {
	
	public class Label : LObject, ISprite {
	
		private const long serialVersionUID = 1L;
	
		private LFont font;
	
		private bool visible;
	
		private int width, height;
	
		private LColor color;
	
		private string label;
	
		public Label(string label, int x, int y):this(LFont.GetDefaultFont(), label, x, y) {
			
		}
	
		public Label(string label, string font, int type, int size, int x, int y):this(LFont.GetFont(font, type, size), label, x, y) {
			
		}
	
		public Label(LFont font, string label, int x, int y) {
			this.font = font;
			this.label = label;
			this.color = LColor.black;
			this.visible = true;
			this.SetLocation(x, y);
		}
	
		public void SetFont(string fontName, int type, int size) {
			SetFont(LFont.GetFont(fontName, type, size));
		}
	
		public void SetFont(LFont font) {
			this.font = font;
		}
	
		public void CreateUI(GLEx g) {
			if (visible) {
				LFont oldFont = g.GetFont();
				uint oldColor = g.GetColorRGB();
				g.SetFont(font);
				g.SetColor(color);
				this.width = font.StringWidth(label);
				this.height = font.GetSize();
				if (alpha > 0 && alpha < 1) {
					g.SetAlpha(alpha);
					g.DrawString(label, X(), Y());
					g.SetAlpha(1.0F);
				} else {
					g.DrawString(label, X(), Y());
				}
				g.SetFont(oldFont);
				g.SetColor(oldColor);
			}
		}
	
		public override int GetWidth() {
			return width;
		}

        public override int GetHeight()
        {
			return height;
		}

        public override void Update(long timer)
        {
	
		}
	
		public RectBox GetCollisionBox() {
			return GetRect(X(), Y(), width, height);
		}
	
		public virtual bool IsVisible() {
			return visible;
		}
	
		public virtual void SetVisible(bool visible) {
			this.visible = visible;
		}
	
		public string GetLabel() {
			return label;
		}
	
		public void SetLabel(int label) {
			SetLabel(label.ToString());
		}
	
		public void SetLabel(string label) {
			this.label = label;
		}
	
		public LColor GetColor() {
			return color;
		}
	
		public void SetColor(LColor color) {
			this.color = color;
		}
	
		public LTexture GetBitmap() {
			return null;
		}
	
		public void Dispose() {
	
		}
	
	}
}
